pinot-minion/src/main/java/org/apache/pinot/minion/taskfactory/TaskFactoryRegistry.java

```diff
@@ -67,19 +71,31 @@ public TaskFactoryRegistry(TaskExecutorFactoryRegistry taskExecutorFactoryRegist
             private final TaskConfig _taskConfig = context.getTaskConfig();
             private final PinotTaskExecutor _taskExecutor = taskExecutorFactory.create();
             private final MinionEventObserver _eventObserver = eventObserverFactory.create();
+            private final MinionMetrics _minionMetrics = MinionContext.getInstance().getMinionMetrics();
 
             @Override
             public TaskResult run() {
-              MinionMetrics minionMetrics = MinionContext.getInstance().getMinionMetrics();
+              HelixManager helixManager = context.getManager();
+              JobContext jobContext = TaskDriver.getJobContext(helixManager, context.getJobConfig().getJobId());
+              // jobContext.getStartTime() return the time in milliseconds of job being put into helix queue.
+              long jobInQueueTime = jobContext.getStartTime();
+              long jobDequeueTime = System.currentTimeMillis();
+              _minionMetrics.addPhaseTiming(taskType, MinionQueryPhase.TASK_QUEUEING, jobDequeueTime - jobInQueueTime);
```

Comment:
One more thing to note is that `MinionQueryPhase.TASK_EXECUTION` logs time in **nanoseconds** while this one logs time in **milliseconds**. The time granularity needs to be unified.

Reply:
helix set the job start time in milliseconds. So if we wanner unify the the granularity of log/gauge, the only option is change the granularity of log as milliseconds.

Reply:
done